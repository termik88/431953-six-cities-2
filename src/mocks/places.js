const places = [
  {
    city: {
      name: `Dusseldorf`,
      location: {latitude: 51.225402, longitude: 6.776314, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`
    ],
    title: `Canal View Prinsengracht`,
    isFavorite: false,
    isPremium: false,
    rating: 3.1,
    type: `room`,
    bedrooms: 1,
    maxAdults: 2,
    price: 101,
    goods: [`Breakfast`, `Laptop friendly workspace`],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
    location: {latitude: 51.248402000000006, longitude: 6.763314, zoom: 16},
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {latitude: 52.37454, longitude: 4.897976, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`
    ],
    title: `House in countryside`,
    isFavorite: false,
    isPremium: true,
    rating: 4.5,
    type: `room`,
    bedrooms: 1,
    maxAdults: 1,
    price: 163,
    goods: [
      `Fridge`,
      `Baby seat`,
      `Laptop friendly workspace`,
      `Breakfast`,
      `Dishwasher`,
      `Towels`,
      `Washer`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
    location: {latitude: 52.389540000000004, longitude: 4.883976, zoom: 16},
    id: 2
  },
  {
    city: {
      name: `Cologne`,
      location: {latitude: 50.938361, longitude: 6.959974, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/12.jpg`
    ],
    title: `Beautiful & luxurious apartment at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 2.1,
    type: `apartment`,
    bedrooms: 1,
    maxAdults: 5,
    price: 254,
    goods: [`Breakfast`, `Laptop friendly workspace`],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `A new spacious villa, one floor. All commodities, jacuzzi and beautiful scenery. Ideal for families or friends.`,
    location: {latitude: 50.930361, longitude: 6.937974, zoom: 16},
    id: 3
  },
  {
    city: {
      name: `Paris`,
      location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `The Joshua Tree House`,
    isFavorite: false,
    isPremium: false,
    rating: 4.3,
    type: `room`,
    bedrooms: 1,
    maxAdults: 2,
    price: 284,
    goods: [`Laptop friendly workspace`],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
    location: {latitude: 48.858610000000006, longitude: 2.330499, zoom: 16},
    id: 4
  },
  {
    city: {
      name: `Amsterdam`,
      location: {latitude: 52.37454, longitude: 4.897976, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg`
    ],
    title: `Beautiful & luxurious apartment at great location`,
    isFavorite: false,
    isPremium: true,
    rating: 4.9,
    type: `hotel`,
    bedrooms: 3,
    maxAdults: 7,
    price: 463,
    goods: [`Laptop friendly workspace`, `Breakfast`],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I rent out a very sunny and bright apartment only 7 minutes walking distance to the metro station. The apartment has a spacious living room with a kitchen, one bedroom and a bathroom with mit bath. A terrace can be used in summer.`,
    location: {latitude: 52.36354, longitude: 4.889976, zoom: 16},
    id: 5
  },
  {
    city: {
      name: `Brussels`,
      location: {latitude: 50.846557, longitude: 4.351697, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`
    ],
    title: `Tile House`,
    isFavorite: false,
    isPremium: false,
    rating: 2.9,
    type: `room`,
    bedrooms: 1,
    maxAdults: 2,
    price: 241,
    goods: [
      `Laptop friendly workspace`,
      `Washer`,
      `Breakfast`,
      `Towels`,
      `Dishwasher`,
      `Fridge`,
      `Baby seat`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
    location: {latitude: 50.867557, longitude: 4.357697, zoom: 16},
    id: 6
  },
  {
    city: {
      name: `Hamburg`,
      location: {latitude: 53.550341, longitude: 10.000654, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/18.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`
    ],
    title: `Penthouse, 4-5 rooms + 5 balconies`,
    isFavorite: false,
    isPremium: false,
    rating: 4.4,
    type: `apartment`,
    bedrooms: 5,
    maxAdults: 7,
    price: 194,
    goods: [
      `Washer`,
      `Air conditioning`,
      `Breakfast`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {latitude: 53.550341, longitude: 9.980654000000001, zoom: 16},
    id: 7
  },
  {
    city: {
      name: `Hamburg`,
      location: {latitude: 53.550341, longitude: 10.000654, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/11.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/18.jpg`
    ],
    title: `Loft Studio in the Central Area`,
    isFavorite: false,
    isPremium: false,
    rating: 3.5,
    type: `room`,
    bedrooms: 1,
    maxAdults: 2,
    price: 215,
    goods: [
      `Washer`,
      `Laptop friendly workspace`,
      `Baby seat`,
      `Breakfast`,
      `Towels`,
      `Air conditioning`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 53.574341000000004,
      longitude: 10.022654000000001,
      zoom: 16
    },
    id: 8
  },
  {
    city: {
      name: `Hamburg`,
      location: {latitude: 53.550341, longitude: 10.000654, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/18.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/5.jpg`
    ],
    title: `The Joshua Tree House`,
    isFavorite: false,
    isPremium: false,
    rating: 3.3,
    type: `house`,
    bedrooms: 2,
    maxAdults: 6,
    price: 519,
    goods: [
      `Towels`,
      `Washer`,
      `Fridge`,
      `Breakfast`,
      `Laptop friendly workspace`,
      `Air conditioning`,
      `Coffee machine`,
      `Dishwasher`,
      `Baby seat`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    location: {
      latitude: 53.558341000000006,
      longitude: 9.999654000000001,
      zoom: 16
    },
    id: 9
  },
  {
    city: {
      name: `Hamburg`,
      location: {latitude: 53.550341, longitude: 10.000654, zoom: 13}
    },
    previewImage: `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
    images: [
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/16.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/13.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/10.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/12.jpg`
    ],
    title: `Beautiful & luxurious apartment at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 2,
    type: `room`,
    bedrooms: 1,
    maxAdults: 3,
    price: 273,
    goods: [
      `Air conditioning`,
      `Washing machine`,
      `Towels`,
      `Dishwasher`,
      `Washer`,
      `Fridge`,
      `Breakfast`,
      `Baby seat`,
      `Coffee machine`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
    location: {
      latitude: 53.528341000000005,
      longitude: 10.018654000000002,
      zoom: 16
    },
    id: 10
  }];

export default places;
