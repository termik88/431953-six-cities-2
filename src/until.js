export const getCitiesList = (placesAll) => [...new Set(placesAll.map((place) => place.city.name))];

export const getRandomCity = (citiesList) => citiesList[Math.floor(Math.random() * Math.floor(citiesList.length))];

const RATING_STARS = 5;
const PERCENT = 100;
const transformRatingToPercent = (ratingPlace) => Math.round(ratingPlace) / RATING_STARS * PERCENT;

export const transformRatingToStarsNumber = (ratingPlace) => ratingPlace * RATING_STARS / PERCENT;

export const prepareUser = (user) => {
  const userData = Object.assign({}, user);
  userData.avatarUrl = userData.avatar_url;
  userData.isPro = userData.is_pro;
  delete userData.avatar_url;
  delete userData.is_pro;

  return userData;
};

export const preparePlace = (places) => {
  const newPlaces = Object.assign({}, places);
  newPlaces.previewImage = newPlaces.preview_image;
  newPlaces.isFavorite = newPlaces.is_favorite;
  newPlaces.isPremium = newPlaces.is_premium;
  newPlaces.maxAdults = newPlaces.max_adults;
  newPlaces.rating = transformRatingToPercent(newPlaces.rating);
  newPlaces.type = newPlaces.type[0].toUpperCase() + newPlaces.type.slice(1);
  delete newPlaces.preview_image;
  delete newPlaces.is_favorite;
  delete newPlaces.is_premium;
  delete newPlaces.max_adults;
  newPlaces.host = prepareUser(newPlaces.host);

  return newPlaces;
};

export const preparePlaces = (placesList) => placesList.map((place) => preparePlace(place));

export const isValidationEmail = (email) => {
  const pattern = /^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/;
  return pattern.test(String(email).toLowerCase());
};

const MONTHS = [
  `January`,
  `February`,
  `March`,
  `April`,
  `May`,
  `June`,
  `July`,
  `August`,
  `September`,
  `October`,
  `November`,
  `December`,
];

export const getMonthYear = (date) => {
  if (!date) {
    return ``;
  }
  const fullDate = new Date(date);
  const month = MONTHS[fullDate.getMonth()];

  return `${month} ${fullDate.getFullYear()}`;
};

const prepareComment = (comment) => {
  const newComment = Object.assign({}, comment);
  newComment.rating = transformRatingToPercent(newComment.rating);
  newComment.commentText = newComment.comment;
  delete newComment.comment;
  newComment.user = prepareUser(newComment.user);

  return newComment;
};

export const prepareComments = (comments) => comments.map((comment) => prepareComment(comment));

export const getPlacesNearest = (currentPoint, allPoints) => {
  const distance = (x1, y1, x2, y2) => Math.sqrt(Math.pow(x2 - x1, 2) + (Math.pow(y2 - y1, 2)));
  const filteredPoints = allPoints.filter((item) => item.city.name === currentPoint.city.name && item.id !== currentPoint.id);
  return filteredPoints
    .map((x) => x)
    .sort((x, y) =>
      Math.abs(distance(x.location.longitude, x.location.latitude, currentPoint.location.longitude, currentPoint.location.latitude))
      - Math.abs(distance(y.location.longitude, y.location.latitude, currentPoint.location.longitude, currentPoint.location.latitude)))
    .slice(0, 3);
};

