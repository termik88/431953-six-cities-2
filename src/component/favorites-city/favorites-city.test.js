import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from "react-router";
import {createBrowserHistory} from "history";

import FavoritesCity from "./favorites-city.jsx";
import {getCitiesList} from "../../until";
import places from "../../mocks/places";

jest.mock(`../favorite-button/favorite-button.jsx`);

describe(`FavoritesCity component test.`, () => {
  it(`Correctly renders after relaunch`, () => {
    const citiesList = getCitiesList(places);

    const mockDate = {
      cityName: citiesList[0],
      favoritesOffers: places,
      path: `/`,
      handleClickCityName: jest.fn()
    };

    const test = renderer
      .create(
          <Router history={createBrowserHistory()}>
            <FavoritesCity
              {...mockDate}
            />
          </Router>).toJSON();

    expect(test).toMatchSnapshot();
  });
});
