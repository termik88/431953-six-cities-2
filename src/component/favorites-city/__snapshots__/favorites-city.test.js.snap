// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesCity component test. Correctly renders after relaunch 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link locations__item-link--test"
        href="/"
        onClick={[Function]}
      >
        <span>
          Dusseldorf
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="place-card place-card--favorites"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__image-wrapper place-card__image-wrapper--favorites"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          <img
            alt="Canal View Prinsengracht"
            className="place-card__image"
            height="200"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/9.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info place-card__info--favorites"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              101
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "3.1%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/1"
            onClick={[Function]}
          >
            Canal View Prinsengracht
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          room
        </p>
      </div>
    </article>
    
    
    
    
    
    
    
    
    
  </div>
</li>
`;
